mopidy/audio/scan.py

```diff
@@ -91,43 +89,44 @@ def _setup_pipeline(uri, proxy_config=None):
 
 def _have_type(element, probability, caps, decodebin):
     decodebin.set_property('sink-caps', caps)
-    struct = gst.Structure('have-type')
-    struct['caps'] = caps.get_structure(0)
-    element.get_bus().post(gst.message_new_application(element, struct))
+    struct = Gst.Structure.new_empty('have-type')
+    struct.set_value('caps', caps.get_structure(0))
+    element.get_bus().post(Gst.Message.new_application(element, struct))
 
 
 def _pad_added(element, pad, pipeline):
-    sink = gst.element_factory_make('fakesink')
+    sink = Gst.ElementFactory.make('fakesink')
     sink.set_property('sync', False)
 
     pipeline.add(sink)
     sink.sync_state_with_parent()
-    pad.link(sink.get_pad('sink'))
+    pad.link(sink.get_static_pad('sink'))
 
-    if pad.get_caps().is_subset(_RAW_AUDIO):
-        struct = gst.Structure('have-audio')
-        element.get_bus().post(gst.message_new_application(element, struct))
+    if pad.query_caps().is_subset(Gst.Caps.from_string('audio/x-raw')):
+        struct = Gst.Structure('have-audio')
+        element.get_bus().post(Gst.Message.new_application(element, struct))
 
 
 def _start_pipeline(pipeline):
-    if pipeline.set_state(gst.STATE_PAUSED) == gst.STATE_CHANGE_NO_PREROLL:
-        pipeline.set_state(gst.STATE_PLAYING)
+    result = pipeline.set_state(Gst.State.PAUSED)
+    if result == Gst.StateChangeReturn.NO_PREROLL:
+        pipeline.set_state(Gst.State.PLAYING)
 
 
 def _query_duration(pipeline):
     try:
-        duration = pipeline.query_duration(gst.FORMAT_TIME, None)[0]
-    except gst.QueryError:
+        duration = pipeline.query_duration(Gst.Format.TIME, None)[0]
```

Comment:
https://lazka.github.io/pgi-docs/#Gst-1.0/classes/Element.html#Gst.Element.query_duration should be `[1]` to get the position.